xbmc/music/MusicThumbLoader.cpp

```diff
@@ -298,6 +298,15 @@ bool CMusicThumbLoader::FillLibraryArt(CFileItem &item)
       {
         if (artitem.artType == "thumb" && !bDiscSetThumbSet)
           item.SetArtFallback(artitem.artType, artname);
+        else if (StringUtils::StartsWith(artitem.artType, "cover"))
+        {
```

Comment:
What are you doing hard coding an art type???? Anyway "Thumb" is the cover.
Need to look at more but this doen't seem right

Reply:
Yeah, perhaps my bad, but "thumb" is hard-coded.  Is fanart1, fanart2 etc not hardcoded then also ?

Reply:
Originally all art types were hard coded, v18 brought revolution that liberated art (meaning that skinners and addons could had a lot more freedom without needing future core changes) but had to leave some things for backwards compatibility. "Thumb" and "fanart" are still there, and there is more to do which will get rid of much of the separate treatment for "fanart". It is all historic. 

You have done pretty much what I did (and probably every dev before us) attempted to copy what is already there. Unfortunately some bits need not to be copied, but nothing is documented so you have no way to know until someone with more experience says "not like that".

I need to take a bit more time over it and see if I can point you in the right direction. It just came as a surprize give all the things in my head, but really no so surprize because there is no reason for you to know otherwise.